models/classes/StatisticalMetadata/Import/Builder/ReportBuilder.php

```diff
@@ -0,0 +1,247 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\StatisticalMetadata\Import\Builder;
+
+use Throwable;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\StatisticalMetadata\Import\Reporter\ImportReporter;
+use oat\tao\model\StatisticalMetadata\Import\Exception\ErrorValidationException;
+use oat\tao\model\StatisticalMetadata\Import\Exception\HeaderValidationException;
+use oat\tao\model\StatisticalMetadata\Import\Exception\WarningValidationException;
+use oat\tao\model\StatisticalMetadata\Import\Exception\AbstractValidationException;
+
+class ReportBuilder
+{
+    public function buildByReporter(ImportReporter $reporter): Report
+    {
+        $reportType = $this->getReportType($reporter);
+
+        $subReport = $this->createReportByResults($reportType, $reporter);
+        $report = $this
+            ->createReportByResults($reportType, $reporter)
+            ->add($subReport);
+
+        $onlyWarningReports = [];
+        $warningAndErrorReports = [];
+
+        /** @var WarningValidationException[]|ErrorValidationException[] $exceptions */
+        foreach ($reporter->getWarningsAndErrors() as $line => $exceptions) {
+            $headerExceptionReports = [];
+            $warningReports = [];
+            $errorReports = [];
+
+            foreach ($exceptions as $exception) {
+                if ($exception instanceof HeaderValidationException) {
+                    $headerExceptionReports[] = $this->createReportByException($exception);
+
+                    continue;
+                }
+
+                if ($exception instanceof WarningValidationException) {
+                    $warningReports[] = $this->createReportByException($exception);
+
+                    continue;
+                }
+
+                if ($exception instanceof ErrorValidationException) {
+                    $errorReports[] = $this->createReportByException($exception);
+                }
+            }
+
+            if (empty($headerExceptionReports) && empty($errorReports) && empty($warningReports)) {
+                continue;
+            }
+
+            $lineMainReportType = empty($headerExceptionReports) && empty($errorReports)
+                ? Report::TYPE_WARNING
+                : Report::TYPE_ERROR;
+            $lineMainReport = Report::create($lineMainReportType, 'line %s: %s', [$line, '']);
+
+            foreach ($headerExceptionReports as $headerExceptionReport) {
+                $lineMainReport->add($headerExceptionReport);
+            }
+
+            foreach ($warningReports as $warningReport) {
+                $lineMainReport->add($warningReport);
+            }
+
+            foreach ($errorReports as $errorReport) {
+                $lineMainReport->add($errorReport);
+            }
+
+            if (empty($errorReports)) {
```

Comment:
You also need to check that you actually have warning reports too, no?

```suggestion
            if (empty($errorReports) && !empty($warningReports)) {
```

Reply:
Nope. This if is to add values inside `$onlyWarningReports` array.